[KB-232] Reason for models and preprocessors (#172)
chhoumann/thesis-chemcam: report_thesis/src/sections/methodology.tex

\subsection{Model and Preprocessing Selection}
Choosing the right models and preprocessing techniques for \gls{libs} data analysis is a challenging task. 

> The literature suggests a variety of models and preprocessing techniques that promise to be adept at handling data that exhibits high-dimensionality, multi-collinearity, and matrix effects.

Reviewer: Should maybe ref back to related works here?

> Additionally, different machine learning models perform best on different oxides in LIBS data due to the complex nature of the data. 

Reviewer: > due to the complex nature of the data.

Needs reasoning?

```suggestion
Additionally, different machine learning models perform better on some oxides than others. 
```

These challenges and model-specific strengths suggests that an optimal approach would involve combining multiple models. 

> This notion is supported by the advent of models such as the \gls{moc}~\cite{cleggRecalibrationMarsScience2017}, which combines the predictions of multiple models using a predefined weighting of the predictions made of each model, per-oxide.

Reviewer: Good!

Also the SuperCam model selection paper, which quite literally uses different models per oxide. Maybe a better ref?

Reviewer: > This approach improved the accuracy over the individual models, but required manual tuning of the weights for each model.

IIRC also the case for the supercam ensembles

This approach improved the accuracy over the individual models, but required manual tuning of the weights for each model.
The manual tuning poses limitations, both in terms of the analysis required to determine a weighting, as well as, the risk for suboptimal weighting.
Based on these considerations, it seems reasonable to consider techniques that could automate such a process, while still being able to leverage the strengths of multiple models.
To address these criteria, we chose to adopt a stacking ensemble approach. 

> As described in Section~\ref{subsec:stacked-generalization}, is a method that utilizes multiple base estimators trained on the same data, whose predictions are then used to train a meta-learner.

Reviewer: ```suggestion
Stacking, as described in Section~\ref{subsec:stacked-generalization}, is a method that utilizes multiple base estimators trained on the same data, whose predictions are then used to train a meta-learner.
```

> This approach has the benefit of being able to leverage the strengths of multiple models on a per-oxide basis in conjunction with the most beneficial preprocessors for each of these models.

Reviewer: Well, that is really a general ensemble benefit for multiple, different regression targets. Not so much a stacking benefit. The benefit of stacking is that it could automate and potentially improve upon the manual tuning by using a meta-learner to discern patterns in the base predictors' outputs.

> We adopted an experimental approach to empirically evaluate the potential of various models and preprocessing techniques, ensuring that our selections were informed by existing literature while also allowing for independent assessment and validation.

Reviewer: I think we may want to explicitly say we reviewed the literature, as we did previously:
> This selection was guided primarily by the literature review but also by exploratory intuition, aiming to discover potentially innovative applications and performances within our specific dataset.

It's hinted at above:
> The literature suggests a variety of models and preprocessing techniques that promise to be adept at handling data that exhibits high-dimensionality, multi-collinearity, and matrix effects.

But I think the part I've highlighted here confuses it a bit. Just gotta clarify that we did a lit review to find models to check, and that this is the way we evaluated them.

> To guide our selection of preprocessing techniques, we had several considerations.

Reviewer: ```suggestion
We had several considerations to guide our selection of preprocessing techniques.
```

Firstly, our review of the literature revealed that there is no consensus on a single, most effective normalization method for \gls{libs} data.
This led us to include the traditional normalization methods, such as Z-score normalization, Min-Max scaling, and Max Absolute scaling, in our experiments.
This approach allowed us to determine which normalization method was most effective for our dataset. 

> Additionally, dimensionality reduction techniques are considered by the literature to be effective techniques for \gls{libs} data due to its high dimensionality. Specifically, \gls{pca} has been widely adopted by the spectroscopic community as an established dimensionality reduction technique~\cite{pca_review_paper}. However, \citet{pca_review_paper} also made the case that the assumptions for \gls{pca} regarding linearity of the data are only met up to a certain point, after which it breaks. They argue that this non-linearity inherent in the data makes \gls{kernel-pca} a valid candidate for \gls{libs} data. Based on their review of the field, and our own review of the literature, not many have studied the effectiveness of \gls{kernel-pca} in the context of \gls{libs} data. Therefore, we decided to include this in our experiments to further assess its potential. In addition to the non-linearity, \citet{pca_review_paper} also argue that the assumptions of normality in the data are not always met in \gls{libs} data. For this reason, we decided to include power transformation and quantile transformation in our experiments as models such as \gls{pca} benefit from a normal distribution of the data. We assume that models such as \gls{pls} may also benefit from a more Gaussian-like distribution of the data, by virtue of the model being partly based on \gls{pca}.

Reviewer: > spectroscopic community

Is that a thing? It sounds weird. Who are they? Can we just say that it's widely adopted in other work / the literature?

Reviewer: Wow, impossible to see in this diff, but all I changed is `However, \citet{pca_review_paper} makes the case that the assumptions`.

Remember to add linebreaks!

```suggestion
Additionally, dimensionality reduction techniques are considered by the literature to be effective techniques for \gls{libs} data due to its high dimensionality. Specifically, \gls{pca} has been widely adopted by the spectroscopic community as an established dimensionality reduction technique~\cite{pca_review_paper}. However, \citet{pca_review_paper} makes the case that the assumptions for \gls{pca} regarding linearity of the data are only met up to a certain point, after which it breaks. They argue that this non-linearity inherent in the data makes \gls{kernel-pca} a valid candidate for \gls{libs} data. Based on their review of the field, and our own review of the literature, not many have studied the effectiveness of \gls{kernel-pca} in the context of \gls{libs} data. Therefore, we decided to include this in our experiments to further assess its potential. In addition to the non-linearity, \citet{pca_review_paper} also argue that the assumptions of normality in the data are not always met in \gls{libs} data. For this reason, we decided to include power transformation and quantile transformation in our experiments as models such as \gls{pca} benefit from a normal distribution of the data. We assume that models such as \gls{pls} may also benefit from a more Gaussian-like distribution of the data, by virtue of the model being partly based on \gls{pca}.
```

Reviewer: This is fantastic otherwise. Really nice you found that paper!

> While these preprocessing techniques are not an exhaustive list, they represent a diverse set of methods.

Reviewer: I think we're severely lacking the section describing our hyperparameter & configuration search. This sentence doesn't make sense if you didn't know that we tried to do a search, in which case it makes sense to want exhaustiveness.

Techniques such as feature selection were not considered to limit the scope of the study and due to time constraints.

> For the selection of models, we also had several requirements.

Reviewer: You write like Yoda sometimes man

```suggestion
We also had several requirements for the selection of models.
```

> The models selected for experimentation had to be diverse to ensure that we had enough breadth in our results to make informed decisions about which models should be included in the final pipeline.

Reviewer: > I think we're severely lacking the section describing our hyperparameter & configuration search.

Author: Agreed. Next task up I suppose.

> Additionally, these models should be suitable for regression tasks and suitable for \gls{libs} data, as evidenced by their \gls{rmsep} and \gls{rmsecv} in the related literature.

Reviewer: > as evidenced by their \gls{rmsep} and \gls{rmsecv} in the related literature.

I'd rather just say that they were being used for LIBS, or adjacent, tasks & that the algorithms are capable of regression tasks. Not so much the RMSEP/RMSECV part. That doesn't serve as evidence. If the paper used RMSEP, it's not unreasonable to assume they already described it as a regression task.

To further bolster our selection pool, we chose to include models that were in the same family of models as those that showed promise in the literature.
For example, we chose to include \gls{etr} and \gls{ngboost} on this basis.

In Table~\ref{tab:preprocessing-models} summarizes the preprocessing techniques and models selected for our experimentation:

\begin{table}[ht]
\centering

> \begin{tabularx}{\columnwidth}{lX} % Define the columns, one fixed-width and one flexible

Reviewer: ```suggestion
\begin{tabularx}{\columnwidth}{lX}
```

\toprule
\multirow{5}{*}{Preprocessing Techniques:} & Z-Score Normalization \\
                                           & Min-Max Normalization \\
                                           & Max Absolute Scaling \\
                                           & Robust Scaling \\
                                           & Norm 3 \\
\midrule
\multirow{6}{*}{Models:}                   & Partial Least Squares \\
                                           & Support Vector Regression \\
                                           & Gradient Boost Regression \\
                                           & Extra Trees Regression \\
                                           & Natural Gradient Boosting \\
                                           & Elastic Nets \\
\bottomrule
\end{tabularx}
\caption{Summary of Preprocessing Techniques and Models}
\label{tab:preprocessing-models}

> \end{table}

Reviewer: There are superb arguments in this section for some of the preprocessing methods, but seemingly none for the models we use.

I'd also consider other preprocessing methods you mentioned (PCA, transforms)... preprocessing methods. Perhaps this table could be more exhaustive, showing how we categorize the items into:
- Transformations methods
- Dimensionality reduction methods
- Normalization / Scaling
- Models

Which are the relevant items we search for + the hyperparameters of each.

Ideally, I think we update the table to contain everything, and add rationale for our use of the normalization methods + models. Especially now that we have the background section to back it!

\subsection{Model and Preprocessing Selection}
Choosing the right models and preprocessing techniques for \gls{libs} data analysis is a challenging task. 
As the literature highlighted in Section~\ref{sec:related-work} suggests, a variety of models and preprocessing techniques that promise to be adept at handling data that exhibits high-dimensionality, multi-collinearity, and matrix effects.
Additionally, different machine learning models perform better on some oxides than others. 
These challenges and model-specific strengths suggests that an optimal approach would involve combining multiple models. 

> This notion is supported by the advent of models such as the \gls{moc}~\cite{cleggRecalibrationMarsScience2017}, which combines the predictions of multiple models using a predefined weighting of the predictions made of each model, per-oxide.

Reviewer 2: ```suggestion
This notion is supported by the advent of models such as the \gls{moc}~\cite{cleggRecalibrationMarsScience2017} model, which combines the predictions of multiple models using a predetermined weighting for each model's predictions on a per-oxide basis.
```

> This approach improved the accuracy over the individual models, but required manual tuning of the weights for each model.
> The manual tuning poses limitations, both in terms of the analysis required to determine a weighting, as well as, the risk for suboptimal weighting.

Reviewer 2: ```suggestion
While this approach improved accuracy compared to individual models, it required manual tuning of the weights for each model.
This manual tuning presents limitations, including the analysis required to determine appropriate weights and the risk of suboptimal weighting.
```

> Based on these considerations, it seems reasonable to consider techniques that could automate such a process, while still being able to leverage the strengths of multiple models.

Reviewer 2: ```suggestion
Given these limitations, it is reasonable to explore techniques that can automate the weighting process while still leveraging the strengths of multiple models.
```

To address these criteria, we chose to adopt a stacking ensemble approach. 
Stacking, as described in Section~\ref{subsec:stacked-generalization}, is a method that utilizes multiple base estimators trained on the same data, whose predictions are then used to train a meta-learner.
This approach has the benefit of being able to leverage the strengths of multiple models on a per-oxide basis in conjunction with the most beneficial preprocessors for each of these models.

We adopted an experimental approach to empirically evaluate the potential of various models and preprocessing techniques, ensuring that our selections were informed by existing literature while also allowing for independent assessment and validation.

We had several considerations to guide our selection of preprocessing techniques.
Firstly, our review of the literature revealed that there is no consensus on a single, most effective normalization method for \gls{libs} data.

> This led us to include the traditional normalization methods, such as Z-score normalization, Min-Max scaling, and Max Absolute scaling, in our experiments.

Reviewer 2: ```suggestion
This led us to include the traditional normalization methods, such as z-score normalization, Min-Max scaling, and Max Absolute scaling, in our experiments.
```
z-score is always lowercase when used mid-sentence in the litterature I've read

While these preprocessing techniques are not an exhaustive list, they represent a diverse set of methods.
Techniques such as feature selection were not considered to limit the scope of the study and due to time constraints.

We also had several requirements for the selection of models.
The models selected for experimentation had to be diverse to ensure that we had enough breadth in our results to make informed decisions about which models should be included in the final pipeline.
Additionally, these models should be suitable for regression tasks and suitable for \gls{libs} data, as evidenced by their \gls{rmsep} and \gls{rmsecv} in the related literature.
To further bolster our selection pool, we chose to include models that were in the same family of models as those that showed promise in the literature.
For example, we chose to include \gls{etr} and \gls{ngboost} on this basis.

> In Table~\ref{tab:preprocessing-models} summarizes the preprocessing techniques and models selected for our experimentation:

Reviewer 2: ```suggestion
Table~\ref{tab:preprocessing-models} summarizes the preprocessing techniques and models selected for our experimentation.
```
